presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetHiveRecordCursor.java

```diff
@@ -337,20 +338,21 @@ public void close()
 
             PrestoReadSupport readSupport = new PrestoReadSupport(useParquetColumnNames, columns, fileSchema);
 
-            List<parquet.schema.Type> fields = columns.stream()
-                    .filter(column -> column.getColumnType() == REGULAR)
-                    .map(column -> getParquetType(column, fileSchema, useParquetColumnNames))
-                    .filter(Objects::nonNull)
-                    .collect(toList());
+            Map<parquet.schema.Type, HiveColumnHandle> typeColumns = new HashMap<>();
```

Comment:
The filter nonNull got removed from this but then the copy of this in `ParquetPageSourceFactory` still does that. Possibly just consolidate this logic into `ParquetTypeUtils` and call it from both places.